src/store/messagesStore.js

```diff
@@ -979,11 +980,15 @@ const actions = {
 			}
 		})
 
-		if (!context.getters.getFirstKnownMessageId(token) || oldestKnownMessageId < context.getters.getFirstKnownMessageId(token)) {
+		if (!context.getters.getFirstKnownMessageId(token) || oldestKnownMessageId < context.getters.getFirstKnownMessageId(token)
+			|| context.getters.getFirstKnownMessageId(token) === MESSAGE.CHAT_BEGIN_ID) {
 			context.dispatch('setFirstKnownMessageId', {
 				token,
 				id: oldestKnownMessageId,
 			})
+			if (context.getters.getVisualLastReadMessageId(token) === MESSAGE.CHAT_BEGIN_ID) {
+				context.dispatch('setVisualLastReadMessageId', { token, id: oldestKnownMessageId })
+			}
 		}
```

Comment:
Could be done before cycle and checked once: oldestKnownMessageId should be positive
```suggestion
		if (context.getters.getVisualLastReadMessageId(token) === MESSAGE.CHAT_BEGIN_ID) {
			do both
		}
```